src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/PreferJaggedArraysOverMultidimensional.cs

```diff
@@ -115,11 +121,12 @@ private static void AnalyzeObjectCreation(OperationAnalysisContext context)
             }
         }
 
-        private static void AnalyzeParameters(SymbolAnalysisContext context, ImmutableArray<IParameterSymbol> parameters)
+        private static void AnalyzeParameters(SymbolAnalysisContext context, ISymbol containingSymbol, ImmutableArray<IParameterSymbol> parameters)
         {
             foreach (IParameterSymbol parameter in parameters)
             {
-                if (IsMultiDimensionalArray(parameter.Type))
+                if (IsMultiDimensionalArray(parameter.Type) &&
+                    !containingSymbol.IsImplementationOfAnyInterfaceMember())
```

Comment:
This "expensive" call is now done for each parameters although the check could be done once. Besides the check was also done previously for the method or property on the calling method.

Reply:
I think that is fine - multidimensional arrays are rare so this is okay. Invoking this for every property/method regardless of the type would be more expensive.